core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -40,6 +43,55 @@
  */
 public abstract class QueryBuilder<QB extends QueryBuilder> extends ToXContentToBytes implements NamedWriteable<QB> {
 
+    public static enum Operator {
```

Comment:
One more thing, I wonder where this new class should go, my preference would be to have it a separate class on this same package, I don't quite love the idea of having an inner class in this abstract class. Do you have a preference?